spec/abstractops.html

```diff
@@ -4,6 +4,28 @@
 <emu-clause id="sec-temporal-abstract-ops">
   <h1>Abstract operations</h1>
 
+  <!-- Based on ECMA-402 9.2.10 GetOption -->
+  <emu-clause id="sec-getoption" aoid="GetOption">
+    <h1>GetOption ( _options_, _property_, _values_, _fallback_ )</h1>
+
+    <p>
+      The abstract operation GetOption extracts the value of the property named _property_ from the provided _options_ object, converts it to a string, checks whether it is one of a List of allowed _values_, and fills in a _fallback_ value if necessary. If _values_ is *undefined*, there is no fixed set of values and any is permitted. If _options_ is *undefined*, then _fallback_ is returned.
+    </p>
+
+    <emu-alg>
+      1. If _options_ is *undefined*, then
+        1. Return _fallback_.
+      1. Set _options_ to ? ToObject(_options_).
```

Comment:
Do we want to throw on `null`? IDL-based APIs don't.

Reply:
Throwing on `null` would be familiar to programmers using defaulting in JS:
```
> function foo({ disambiguation = 'bar' } = {}) { console.log(disambiguation); }
> foo()
bar
> foo(null)
TypeError: Cannot destructure property `disambiguation` of 'undefined' or 'null'.
```
This would be the first options bag API in ECMA-262, but ECMA-402 (where I copied this algorithm from) also seems to throw on null:
```
> new Intl.DateTimeFormat('en', null)
TypeError: Cannot convert undefined or null to object
```

Reply:
The language most frequently treats undefined and null the same (default arguments does not follow this pattern); i'd prefer to stick to nullish coalescing/optional chaining/the more common pattern of not throwing on null (unless you also throw on undefined)